assert/assertions.go

```diff
@@ -407,6 +407,9 @@ func NotNil(t TestingT, object interface{}, msgAndArgs ...interface{}) bool {
 	if h, ok := t.(tHelper); ok {
 		h.Helper()
 	}
+	if isNonNillableType(object) {
```

Comment:
we should make similar changes to `Nil`, keeping the contract `!Nil(anyType) == NotNil(anyType)` same.
also we could log it, instead of failing the test. i left a comment in #570

Reply:
I don't see any reason this should be a log. If the function is called with a value that is not nil, it should fail the test. That's the entire purpose of the assertion.

Reply:
I agree with @dnephin - incorrectly used assertions should not work.

Reply:
I think failing is OK.